### ISSUE TITLE
Newly created packages should be discovered automatically.

### ISSUE THREAD

**Issue by dmitshur** [shurcooL/home#19]
Currently, code discovery happens only once, during process startup:

https://github.com/shurcooL/home/blob/9c781c585b000571fbc577aaada13b8a509b83f8/main.go#L187-L192

https://github.com/shurcooL/home/blob/9c781c585b000571fbc577aaada13b8a509b83f8/internal/code/code.go#L63-L64

No rediscovery is performed afterwards. As a result, any new Go packages that are created while the process is running do not appear until the process is restarted. Similarly, removed Go packages do not disappear until restart.

In order to improve the user experience, they should be discovered automatically. Rediscovery should happen when:

- a new git commit is pushed that contains new Go packages (or deletes some)
- a new repository is created

At this time, there is no general-purpose mechanism to move or delete repositories. Because they're very rare, such operations will continue to be done ad hoc manually and are considered out of scope for this issue for now.

**Comment by dmitshur** [comment]
Made some progress on automatic package discovery on push:

![image](https://user-images.githubusercontent.com/1924134/46929123-f6490b80-d00b-11e8-9539-720271e76717.png)

It needs more work, tests, cleanup.
